autowarefoundation/autoware
# [Feature] region tlr mxnet

## Status
**PRODUCTION / DEVELOPMENT**

## Description
This PR implements the issue described in autowarefoundation/autoware_ai#32 

## Related PRs
autowarefoundation/autoware#930 autowarefoundation/autoware#953 

## Todos
  - [X] Tested on Ubuntu 14.04, CUDA8, CUDNN5.1
    - [X] ResNet
    - [X] MobileNet
    - [X] DenseNet
  - [ ] Tested on  Ubuntu 16.04, CUDA8, CUDNN5.1


## Steps to Test or Reproduce
Prerequisites:
- VectorMap loaded, 
- PCD Map loaded, 
- Localization working (NDT), 
- Intrinsic and Extrinsic Camera Calibration being published, 
- tf between camera and lidar.
- MxNet with the C++ package installed. Please follow detailed instruction on the `README.md` 
https://github.com/CPFL/Autoware/blob/feature/region_tlr_mxnet/ros/src/computing/perception/detection/packages/road_wizard/nodes/region_tlr_mxnet/README.md

1. Launch `feat_proj` from a sourced terminal.
  - `feat_proj` can find the traffic lights of only the current lane if **VectorMapServer** (*Database* Tab) is running and the node is launched with the `use_path_info` param set to *true*.  This required `/final_waypoints` to be published (check Autoware `velocity_set` module).
  `roslaunch road_wizard feat_proj.launch camera_id:=/camera0 use_path_info:=true`
2. Launch tlr_node from a terminal:
  `rosrun road_wizard region_tlr_mxnet _image_raw_topic:=/camera0/image_source _network_definition_file:=/PATH/TO/FILE/network_definition.json _pretrained_model_file:=/PATH/TO/FILE/pretrained_weights.params _use_gpu:=true` <- change according
